lib/rspec/matchers/built_in/match.rb

```diff
@@ -5,6 +5,22 @@ module BuiltIn
       # Provides the implementation for `match`.
       # Not intended to be instantiated directly.
       class Match < BaseMatcher
+        # Used to specify the captures we match against
+        # Accepts variadic non-named captures
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          extend WithCaptures
+        end
```

Comment:
Cool. Though this fits with my use case (testing a regexp), I'm still wondering if it makes sense to have `with_captures` available on a general `Match`.  The actual could be a string that is expected to match a regex.

Per @myronmarston's example:

```
expect("2015-01-01").to match(/(\d{4})-/).with_captures("2015")
```